bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/WDChannels.java

```diff
@@ -44,4 +59,117 @@ public synchronized WorkerDataChannel getWorkerDataChannel(String name) {
         }
         return channel;
     }
+
+    public Object receiveDataMultipleChannels(Strand strand, ReceiveField[] receiveFields, Type targetType)
+            throws Throwable {
+        if (strand.workerReceiveMap == null) {
+            strand.workerReceiveMap = ValueCreator.createMapValue(targetType);
+        }
+        for (ReceiveField field : receiveFields) {
+            WorkerDataChannel channel = getWorkerDataChannel(field.channelName());
+            if (!channel.isClosed()) {
+                Object result = channel.tryTakeData(strand, true);
+                checkAndPopulateResult(strand, field, result, channel);
+            } else {
+                if (channel.getState() == WorkerDataChannel.State.AUTO_CLOSED) {
+                    checkAndPopulateResult(strand, field, ErrorUtils.createNoMessageError(field.channelName()),
+                            channel);
+                }
+            }
```

Comment:
HBT? can avoid the nested if and duplicate lines.
```suggestion
            WorkerDataChannel.State state = channel.getState();

            Object result = null;
            if (state == WorkerDataChannel.State.OPEN) {
                result = channel.tryTakeData(strand, true);
            } else if (state == WorkerDataChannel.State.AUTO_CLOSED) {
                result = ErrorUtils.createNoMessageError(field.channelName());
            }

            if (result != null) {
                checkAndPopulateResult(strand, field, result, channel);
            }
```

Reply:
We can't depend on the `result != null` for handling the result because the strand yielding is managed based on it inside the `checkAndPopulateResult` method. I can still refactor it like this.
```suggestion
            WorkerDataChannel.State state = channel.getState();
            Object result = null;
            switch (state) {
                case OPEN:
                    result = channel.tryTakeData(strand, true);
                    break;
                case CLOSED:
                    continue;
                case AUTO_CLOSED:
                    result = ErrorUtils.createNoMessageError(field.channelName());
                    break;
            }
            checkAndPopulateResult(strand, field, result, channel);
```

Reply:
ah yes. +1 on your suggestion.